compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -843,27 +771,16 @@ impl<'a> Resolver<'a> {
                 err
             }
             ResolutionError::CannotCaptureDynamicEnvironmentInFnItem => {
-                let mut err = struct_span_err!(
-                    self.session,
+                self.session.create_err(errs::CannotCaptureDynamicEnvironmentInFnItem {
                     span,
-                    E0434,
-                    "{}",
-                    "can't capture dynamic environment in a fn item"
-                );
-                err.help("use the `|| { ... }` closure form instead");
-                err
+                    closure_hint: "|| { ... }",
```

Comment:
Why can't this be in the Fluent message?

Reply:
fluent parser was failing on this closure string for some reason. I think it was due to '{'. I would need to look into in again.

Reply:
Similar thing is failing to parse at another place.
![image](https://user-images.githubusercontent.com/1040077/188287355-e02f47f2-cd03-4849-b9dc-a99958033c1c.png)
For now I am passing just code related info from struct field.
Please let me know if there is something which could be done about it. Maybe I am missing something.

Reply:
Use `{"{"}` to escape `{`, and similarly for `}`. https://projectfluent.org/fluent/guide/special.html